src/effects/effectslot.cpp

```diff
@@ -2,258 +2,528 @@
+        // Do not delete the slots; clear the parameters from the slots
+        // The parameter slots are used by the next effect, but the EffectParameters
+        // are deleted below.
+        for (auto pSlot : std::as_const(slotList)) {
+            pSlot->clear();
+        }
+    }
+    for (auto& parameterList : m_allParameters) {
+        parameterList.clear();
+    }
+    for (auto& parameterList : m_loadedParameters) {
+        parameterList.clear();
     }
-    for (const auto& pParameter : qAsConst(m_buttonParameters)) {
-        pParameter->clear();
+    for (auto& parameterList : m_hiddenParameters) {
+        parameterList.clear();
     }
-    m_pEffect.clear();
-    emit updated();
+
+    m_pManifest.clear();
+
+    removeFromEngine();
+}
+
+void EffectSlot::loadParameters() {
+    //qDebug() << this << m_group << "loading parameters";
+    int numTypes = static_cast<int>(EffectParameterType::NUM_TYPES);
+    for (int parameterTypeId = 0; parameterTypeId < numTypes;
+            ++parameterTypeId) {
+        const EffectParameterType parameterType =
+                static_cast<EffectParameterType>(parameterTypeId);
+
+        m_pControlNumParameters[parameterType]->forceSet(
+                numParameters(parameterType));
+
+        int slot = 0;
+        const auto& loadedParameters = m_loadedParameters.value(parameterType);
+        for (const auto& pParameter : loadedParameters) {
+            // LV2 effects may have more parameters than there are slots available
+            if ((unsigned)slot >= kDefaultMaxParameters) {
+                return;
+            }
+            VERIFY_OR_DEBUG_ASSERT(
+                    slot <= m_parameterSlots.value(parameterType).size()) {
+                break;
+            }
+            m_parameterSlots.value(parameterType)
+                    .at(slot)
+                    ->loadParameter(pParameter);
+            slot++;
+        }
+
+        // Clear any EffectParameterSlots that still have a loaded parameter from before
+        // but the loop above did not load a new parameter into them.
+        for (; slot < m_parameterSlots.value(parameterType).size(); slot++) {
+            m_parameterSlots.value(parameterType).at(slot)->clear();
+        }
+
+        m_hiddenParameters[parameterType].clear();
+        const auto& allParameters = m_allParameters.value(parameterType);
+        for (const auto& pParameter : allParameters) {
+            if (!m_loadedParameters.value(parameterType).contains(pParameter)) {
+                m_hiddenParameters[parameterType].append(pParameter);
+            }
+        }
+    }
+}
+
+void EffectSlot::hideParameter(EffectParameterPointer pParameter) {
+    auto parameterType = pParameter->manifest()->parameterType();
+    VERIFY_OR_DEBUG_ASSERT(
+            m_allParameters.value(parameterType).contains(pParameter)) {
+        return;
+    }
+    VERIFY_OR_DEBUG_ASSERT(
+            !m_hiddenParameters.value(parameterType).contains(pParameter)) {
```

Comment:
It is not exclusive to LV2 effects. I can reproduce this sometimes when toggling the hidden state of a parameter with any effect. I'm looking into it...

Reply:
Have you figured out how to reproduce the debug assertion? I have not figured out how to trigger it reliably...

Reply:
Actually I think it was the redundant connections. I can't reproduce the debug assertion anymore since 89b8247.

Reply:
Unfortunately not reliably. I can provoke it by rapidly hiding and unhiding a parameter and the debug assertion will be triggered once every 20 attempts or so, but I don't know what sequence of events comes before it, I only know that its triggered in the handler in weffectchainpresetbutton.cpp

Reply:
I've tried hiding and unhiding parameters many times now and I haven't been able to trigger the debug assertion since 89b8247...

Reply:
Ok so I figured it out. Before your patch, clicking in the menu, but outside the label or the checkbox triggered the handler as well. Since the debug assertion triggered and the function returned early, nothing happened, the menu just close itself. 
IMO this is still a kind of a bug because it leads to poor UX where the user clicked on a menu entry, but instead of doing the associated action, the menu just closes itself like if the user just clicked somewhere outside the menu entirely. So ideally instead of removing the handler, we should try to fix it or implement the menu appropriately so it works like most other menus in mixxx